### DISCUSSION TITLE
Basic Include files error

### DISCUSSION THREAD

**Discussion by naman0411** [microsoft/vscode-cpptools#12153] [Q&A]
Hi, I recently redownloaded c++ 13.1 using msys2 on my computer and know vs code is giving me errors in basic include lines like #include <iostream>.
I added the appropriate path variables - "C:\msys64\ucrt64\bin"  and tried changing the intellisense configurations as mentioned in various places online to no avail.
![image](https://github.com/microsoft/vscode-cpptools/assets/117378576/c7c4cd3a-6294-4788-81ed-5db7e8a655bf)

g++ -o demo demo.cpp does not create any file.

g++ -v -o demo demo.cpp has the following output.
![image](https://github.com/microsoft/vscode-cpptools/assets/117378576/87d95e5d-019a-4fb4-934f-373474709663)

As seen the g++ is not doing any search for #include files.

What should I do?

**Comment by sean-mcmanus** [comment]
@naman0411 We need more info -- what does the red error squiggle say in the message? What is your c_cpp_properties.json? Have you set your compilerPath property in that file? We have some tutorials at https://code.visualstudio.com/docs/cpp/config-mingw .

**Reply by naman0411** [reply]
![image](https://github.com/microsoft/vscode-cpptools/assets/117378576/399e76f3-b7a3-4848-a516-9b163be4485f)
This is my c_cpp_properties.json. The compiler path is the same one I see when i run g++ -v.
![image](https://github.com/microsoft/vscode-cpptools/assets/117378576/1d2c4427-a7a8-4195-aa96-ae56d6e8926d)


Red error squiggle shows cannot open source file "iostream". Please run the 'Select IntelliSense Configuration...' command to locate your system headers.C/C++(1696). But any Configuration I select shows the same error.

I saw on some other forums that we have to add other includePaths in the json file which are seen on running  g++ -E -x c++ -v - but in my terminal no include path search is being shown.
![image](https://github.com/microsoft/vscode-cpptools/assets/117378576/c4bcc355-e3b5-4910-b084-66ce11360c53)

I followed the tutorial you linked during installation still the issue persists.

**Reply by sean-mcmanus** [reply]
@naman0411 Does your build work? What do you see in the "Includes:" section when you run the C/C++: Log Diagnostics command with the source file active? I think you should see `C:\msys64\ucrt64\include\c++\13.1.0`. You may also want to verify that that path exists.

**Reply by naman0411** [reply]
No, the code won't compile. 
The Includes: does not show the path you mentioned in the current configuration of the extension. If i manually put the path "C:\msys64\ucrt64\include\c++\13.1.0/**" in Include path then a different error shows up at the same place.  
"cannot open source file "locale.h" (dependency of "iostream"). Please run the 'Select IntelliSense Configuration...' command to locate your system header"
Now in Log Diagnostics 
![image](https://github.com/microsoft/vscode-cpptools/assets/117378576/cdf1e128-ed1d-4b18-8528-0e1da99d6b6b)
is shown but below some potential errors are also displayed.
![image](https://github.com/microsoft/vscode-cpptools/assets/117378576/ef8465da-f115-4a5c-9376-5abe7a3d8ed0)

**Reply by sean-mcmanus** [reply]
It sounds like your mingw compiler isn't installed correctly. We're supposed to get the system include paths from querying the compiler. If you set C_Cpp.loggingLevel to debug it might show info on why the compiler query is failing. You shouldn't use "**" recursive includes with system includes as the order is likely to be incorrect, but you could try using includePaths like.

        "C:\msys64\ucrt64\include\c++\13.1.0",
        "C:\msys64\ucrt64\include\c++\13.1.0\x86_64-w64-mingw32",
        "C:\msys64\ucrt64\include\c++\13.1.0\backward",
        "C:\msys64\ucrt64\lib\gcc\x86_64-w64-mingw32\13.1.0\include",
        "C:\msys64\ucrt64\include",
        "C:\msys64\ucrt64\lib\gcc\x86_64-w64-mingw32\13.1.0\include-fixed"

**Reply by naman0411** [reply]
This is what i get on setting debug logging level after changing c_cpp_properties back to default.
![image](https://github.com/microsoft/vscode-cpptools/assets/117378576/77e71621-b26d-4efc-8939-a9b9762c3532)


On changing the paths to the ones you specified I then get errors like cout isn't defined and argument list for variable template "endl" is missing.
Output shown is 
![image](https://github.com/microsoft/vscode-cpptools/assets/117378576/bb13ed72-1179-4f62-8288-891be57eb6ef)


I installed the mingw compiler from the link specified in the tutorial. Should I try from elsewhere?

**Comment by browntarik** [comment]
1) Could you show the output of g++ -o demo demo.cpp without verbose logging? This will tell if the code is actually compiling or not.

2) It looks like you are using the incorrect IntelliSense mode "windows-msvc-64" while you are using the G++ compiler. To fix this, run the C/C++: Edit Configurations (UI) command to change the IntelliSense mode to "windows-gcc-x64"
![image](https://github.com/microsoft/vscode-cpptools/assets/111317156/3d0bc85c-95fe-4f84-bd89-27e99568a066)

**Reply by naman0411** [reply]
1) It is not producing any output. There is no file created so compiling issue is present.
2) I did this but still the same issue persists.
